### PR TITLE
[#11797] Communities issues

### LINKED ISSUE #11797: Communities issues
Umbrella issue for leftovers from https://github.com/status-im/status-react/pull/11566

### 1. Reply option is shown on read-only chats on long tap on messages

 **Steps:**
 - as user A: create community (require approval), create chat and send some message
- as user B: import community, join
- as user B: long tap on message in chat
**Expected result:**
Reply is not available
 **Actual result:**
IMAGE 2021-02-22 15:55:56

 ### 2. Can't set image from gallery as community icon (IOS only)
https://user-images.githubusercontent.com/4557972/108726643-f173bb00-7527-11eb-84d9-1546d816a9e7.mp4

 ### 3. Navigation: back from chat brings you to the chat list instead of chats inside community
 **Steps:**
 - as user A: create community, create a chat inside it, send several messages
- as user A: tap back
**Expected result:**
community with chats
 **Actual result:**
general chat list

 ### 4. Navigation: leave community from "Members" page doesn't redirect you to chat list
 **Steps:**
 - as user A: create community 
- as user A: open ... > tap on community name
- as user A: tap on leave community
**Expected result:**
redirected to home view
 **Actual result:**
need to go back manually

### 5. Placeholder for empty community says about + button 
Should be replaced 
IMAGE 2021-02-22 16:01:59

### 6. For community without picture icon colors are different 
 **Steps:**
 - as user A: create community 
- as user A: open ... 
Monosnap 2021-02-22 16-02-41

### 7. Invitations in 1-1 chat: preview - upgrade to latest version to see community preview, custom icons are not shown, the description doesn't fit message bubble

 **Steps:**
 - as user A: create community 
- as user A: open ... > Share > send it to User B (should be added as your contact)
- as user B: check out the invite

Monosnap 2021-02-22 16-12-40
Custom image / description:
Monosnap 2021-02-22 16-12-40
Preview
Monosnap 2021-02-22 16-13-55

### 8: no outgoing invite message is shown in 1-1 chat if sharing it to  user via public key
**Steps:**
1)  create community (no membership requirement)
2) options -> share
3) paste valid public key
4) tap invite

**Expected result:**
outgoing invite in 1-1 chat 
**Actual result:**
outgoing invite in 1-1 chat is not shown in chat view, but it is shown in preview

### 9. "No membership" by default when creating community 
https://github.com/status-im/status-react/pull/11566#issuecomment-785270618

## Acceptance criteria

Fixed following issues

- [ ] 1. Reply option is shown on read-only chats on long tap on messages
- [ ] 2. Can't set image from gallery as community icon (IOS only)
- [ ] 3. Navigation: back from chat brings you to the chat list instead of chats inside community
- [ ] 4. Navigation: leave community from "Members" page doesn't redirect you to chat list
- [ ] 5. Placeholder for empty community says about + button
- [ ] 6. For community without picture icon colors are different
- [ ] 7. Invitations in 1-1 chat: preview - upgrade to latest version to see community preview, custom icons are not shown, the description doesn't fit message bubble
- [ ] 8. no outgoing invite message is shown in 1-1 chat if inviting user via public key
- [ ] 9. "No membership" by default when creating community

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #11797

### CODE REVIEW

**Comment by churik** [comment]
@flexsurfer thank you for PR!
I will post here possible improvements and issues, and you will decide what can be fixed here and what is worth to fix / log  separately.

### ISSUE 1: When creating community, by default Membership requirement is "No Membership", but it is not shown in Membership requirement screen

**Steps:**
1) Communities > tap on Create community
2) Tap on membership requirement
Monosnap 2021-03-10 11-29-47

### ISSUE 2: Several public channels are creating when tapping "Create" several times
Also question here: don't see any limitations for public chat name, but I believe it should be the same as for public chats (may be wrong), but using long name/links can lead to new issues.
**Steps:**
1) Communities > Create community
2) Options > Create a channel 
3) Enter channel name > Tap on "Create" several times

IMAGE 2021-03-10 11:46:48

### ISSUE 3: Would be awesome to have QR on share community screen
1) Communities > Create community
2) Options > Share
Ideally, this field should be like home > + > start new chat (able to handle ENS, chat keys, QR).
Btw, on this screen also IMO better to use "chat key" instead of "user public key" and at least have read QR code option
IMAGE 2021-03-10 11:58:00

**Comment by churik** [comment]
### ISSUE 4: No picture in invitation to community
**Steps:**
1) Communities > Create community
2) Options > Share > Share with contacts
3) Check 1-1 chat for invitation message
https://user-images.githubusercontent.com/4557972/110620789-bc0ed480-8199-11eb-9a5f-810e6e7c545e.mp4

### ISSUE 5: No way to see full description of community
**Steps:**
1) Communities > Create community with long description
2) Open + > Communities
3) Tap on community
4) Tap options > Community info
Monosnap 2021-03-10 14-50-13
Monosnap 2021-03-10 15-25-03

### ISSUE 6: Unread messages are shown as for 1-1 chats
Monosnap 2021-03-10 15-08-09
IMO may be confusing for users, as we shown messages in counter in case they are in 1-1 or group chat.
would be nice to have opinion of @hesterbruikman 

### ISSUE 7: Would be nice to have QR codes in import/export private community keys screens
IMAGE 2021-03-10 15:41:12
IMAGE 2021-03-10 15:41:16

### ISSUE 8: No validation in "Import community" and "Share" screens
When entering invalid value and confirm, nothing happens. Expected that validation error will be shown.
IMAGE 2021-03-10 15:47:15

### ISSUE 9: Can tap on "Share" several times and several invitations will be sent
**Steps:**
1) Communities > Create community
2) Open + > Communities
3) Tap on community
4) Tap options > Share > Select contact > tap on Share several times

### ISSUE 10: own ENS name is not shown in Members
ENS name should be set in Profile.
**Steps:**
1) Communities > Create community (Require invitation)
2) Options > Community info > Members
IMAGE 2021-03-10 16:09:15

**Comment by churik** [comment]
### QUESTION 11: what about gaps ("Fetch messages") and Fetch more messages (previous history?)
For proper testing I need to have communities created > 2 days ago and one gap in history > 24h.
Would be nice to have build to test this behavior 

### ISSUE 12: Invitation to "Require approval" community is not visible for sender in 1-1 chat after sending

**Steps:**
1) Communities > Create community ("Require approval")
2) open community > Options > invite contact 
3) open 1-1 chat with invited contact

**Expected result:** Invite is visible 

**Actual result:** Invite is visible in preview, but not in chat view
After relogin message appears.

**OS:** IOS, Android

https://user-images.githubusercontent.com/4557972/110766932-fd64ba00-8255-11eb-8b6b-9eff2bf561f9.mp4

### QUESTION 13: no clear explanation for the user about types of communities
IMO info on this screen is not enough to understand https://notes.status.im/testing-communities this behavior.

IMAGE 2021-03-11 10:44:44
cc @hesterbruikman 

### ISSUE 14: No Nickname is shown if tap on User from Members screen, on membership request screnn
**Steps:**
1) Communities > Create community ("Require approval")
2) open community > Options > share chat with user B (in your contact list and has custom nickname) 
3) open membership request > accept it
4) open community > Members > Tap on options on member with nickname

Monosnap 2021-03-11 10-51-28
Monosnap 2021-03-11 12-14-41

### ISSUE 15: Membership request deny is not visible
**Steps:**
1) admin user: Communities > Create community ("Require approval") > invite contact 
2) invited: accept invite > share community with another user
3) another user: send membership request
4) admin user: deny membership request

**Expected result:** another user can see that access is denied
**Actual result:**  "Request is pending"

**OS:** IOS, Android

### ISSUE 16: Channel icons have different colors if open community from invite
**Steps:**
1) admin user: Communities > Create community ("Require approval") > invite contact 
2) invited: Open invite message several times
https://user-images.githubusercontent.com/4557972/110777254-38201f80-8261-11eb-9839-6bc40687332c.mp4

### ISSUE 17: Invite message is the same for sender and receiver
1) admin user: Communities > Create community ("Require approval") > invite contact 

photo_2021-03-11 12 05 47 jpeg 2021-03-11 12-06-46

**Comment by churik** [comment]
### (ISSUE or FEATURE) 18: After kicking member and joining no messages in  chats
not sure, maybe it is expected behavior
**Steps:**
1) admin user: Communities > Create community ("Require approval") > create chat inside community 
2) admin user: invite contact 
3) invited user: request access
4) admin user: approve access > kick invited user
5) invited user: re-request access
6) admin user: approve access 
7) invited user: join community 

**Expected result:** messages for chat are visible

**Actual result:** No messages 

**OS:** IOS, Android

### ISSUE 19: Mess with share and invite options in different types of community
**For "Require invite" option:**
- "Invite" is available in community > options
IMAGE 2021-03-11 15:13:48
- "Share" is available in community > community info screen
IMAGE 2021-03-11 15:14:20
- "Invite" is available in community > community info screen > members
**For "Require approval" option:**
- "Invite" and "Share" are available in community > options
IMAGE 2021-03-11 15:23:49

**Actual result:** Icons for both options are the same, some options are not available in community > options
**Expected result:** both options in community > options, icons are different

### ISSUE 20: User is not shown as member when he imports the community
**Expected result:** user is shown as a member
**Actual result:** user is not shown as a member and he can kick admin

**Comment by flexsurfer** [comment]
BUGS:

- [x]  ~ISSUE 1: When creating community, by default Membership requirement is "No Membership", but it is not shown in Membership requirement screen~
- [x] ~ISSUE 2: Several public channels are creating when tapping "Create" several times~
- [x] ~ISSUE 4: No picture in invitation to community~
- [x] ~ISSUE 5: No way to see full description of community~
- [x] ~ISSUE 6: Unread messages are shown as for 1-1 chats~
- [x] ~ISSUE 9: Can tap on "Share" several times and several invitations will be sent~
- [x] ~ISSUE 10: own ENS name is not shown in Members~
- [x] ~ISSUE 12: Invitation to "Require approval" community is not visible for sender in 1-1 chat after sending~
- [x] ~ISSUE 14: No Nickname is shown if tap on User from Members screen, on membership request screnn~
- [x] ~ISSUE 16: Channel icons have different colors if open community from invite~
- [x] ~ISSUE 17: Invite message is the same for sender and receiver (by design, see figma)~

OTHER:
ISSUE 3: Would be awesome to have QR on share community screen
ISSUE 7: Would be nice to have QR codes in import/export private community keys screens
ISSUE 8: No validation in "Import community" and "Share" screens
QUESTION 11: what about gaps ("Fetch messages") and Fetch more messages (previous history?)
QUESTION 13: no clear explanation for the user about types of communities
(ISSUE or FEATURE) 18: After kicking member and joining no messages in chats
ISSUE 19: Mess with share and invite options in different types of community
 ISSUE 20: User is not shown as member when he imports the community

For later:
ISSUE 15: Membership request deny is not visible (Andrea's comment: that's for later, not sure even we want to theoretically)

**Comment by churik** [comment]
@cammellos 
I believe we can merge this part for now (it is definitely better then before, but still have issues/questions and doesn't  match Figma mostly) and continue step-by-step, what do you think?

**Comment by cammellos** [comment]
@churik sure, thanks a lot @flexsurfer !

File: src/status_im/communities/core.cljs [hunk 1]
```diff
@@ -306,23 +292,23 @@
...
             (navigation/navigate-back)
-            (handle-response response)))
+            (handle-response response-js)))
 
 (fx/defn open-create-community
   {:events [::open-create-community]}
   [{:keys [db] :as cofx}]
   (fx/merge cofx
-            {:db (assoc db :communities/create {})}
+            {:db (assoc db :communities/create {:membership 1})}
```
Comment (cammellos) [inline src/status_im/communities/core.cljs]: we should use the one defined in `constants`

File: src/status_im/ui/screens/communities/create.cljs [hunk 2]
```diff
@@ -29,14 +30,20 @@
                 :height               crop-size})
 
 (defn pick-pic []
-  (react/show-image-picker
-   #(>evt [::communities/create-field :image (.-path ^js %)])
-   crop-opts))
+  (js/setTimeout
```
Comment (cammellos) [inline src/status_im/ui/screens/communities/create.cljs:34]: Would you mind leaving a comment on why we need a timeout? It's not immediately clear to the untrained eye (myself :) )

**Review by cammellos — APPROVED** [review]
just a couple of minor comments, thanks!

**Pull request merged**